bundles/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/internal/common/LcnChannelGroup.java

```diff
@@ -0,0 +1,117 @@
+ */
+package org.openhab.binding.lcn.internal.common;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.lcn.internal.subhandler.AbstractLcnModuleSubHandler;
+import org.openhab.binding.lcn.internal.subhandler.LcnModuleBinarySensorSubHandler;
+import org.openhab.binding.lcn.internal.subhandler.LcnModuleCodeSubHandler;
+import org.openhab.binding.lcn.internal.subhandler.LcnModuleKeyLockTableSubHandler;
+import org.openhab.binding.lcn.internal.subhandler.LcnModuleLedSubHandler;
+import org.openhab.binding.lcn.internal.subhandler.LcnModuleLogicSubHandler;
+import org.openhab.binding.lcn.internal.subhandler.LcnModuleOutputSubHandler;
+import org.openhab.binding.lcn.internal.subhandler.LcnModuleRelaySubHandler;
+import org.openhab.binding.lcn.internal.subhandler.LcnModuleRollershutterOutputSubHandler;
+import org.openhab.binding.lcn.internal.subhandler.LcnModuleRollershutterRelaySubHandler;
+import org.openhab.binding.lcn.internal.subhandler.LcnModuleRvarLockSubHandler;
+import org.openhab.binding.lcn.internal.subhandler.LcnModuleRvarSetpointSubHandler;
+import org.openhab.binding.lcn.internal.subhandler.LcnModuleS0CounterSubHandler;
+import org.openhab.binding.lcn.internal.subhandler.LcnModuleThresholdSubHandler;
+import org.openhab.binding.lcn.internal.subhandler.LcnModuleVariableSubHandler;
+
+/**
+ * Defines the supported channels of an LCN module handler.
+ *
+ * @author Fabian Wolter - Initial contribution
+ */
+@NonNullByDefault
+public enum LcnChannelGroup {
+    OUTPUT(4, LcnModuleOutputSubHandler.class),
+    ROLLERSHUTTEROUTPUT(1, LcnModuleRollershutterOutputSubHandler.class),
+    RELAY(8, LcnModuleRelaySubHandler.class),
+    ROLLERSHUTTERRELAY(4, LcnModuleRollershutterRelaySubHandler.class),
+    LED(12, LcnModuleLedSubHandler.class),
+    LOGIC(4, LcnModuleLogicSubHandler.class),
+    BINARYSENSOR(8, LcnModuleBinarySensorSubHandler.class),
+    VARIABLE(12, LcnModuleVariableSubHandler.class),
+    RVARSETPOINT(2, LcnModuleRvarSetpointSubHandler.class),
+    RVARLOCK(2, LcnModuleRvarLockSubHandler.class),
+    THRESHOLDREGISTER1(5, LcnModuleThresholdSubHandler.class),
+    THRESHOLDREGISTER2(4, LcnModuleThresholdSubHandler.class),
+    THRESHOLDREGISTER3(4, LcnModuleThresholdSubHandler.class),
+    THRESHOLDREGISTER4(4, LcnModuleThresholdSubHandler.class),
+    S0INPUT(4, LcnModuleS0CounterSubHandler.class),
+    KEYLOCKTABLEA(8, LcnModuleKeyLockTableSubHandler.class),
+    KEYLOCKTABLEB(8, LcnModuleKeyLockTableSubHandler.class),
+    KEYLOCKTABLEC(8, LcnModuleKeyLockTableSubHandler.class),
+    KEYLOCKTABLED(8, LcnModuleKeyLockTableSubHandler.class),
+    CODE(0, LcnModuleCodeSubHandler.class);
+
+    private int count;
+    private Class<? extends AbstractLcnModuleSubHandler> subHandlerClass;
+
+    private LcnChannelGroup(int count, Class<? extends AbstractLcnModuleSubHandler> subHandlerClass) {
+        this.count = count;
+        this.subHandlerClass = subHandlerClass;
+    }
```

Comment:
Here is an example of what I mean:
```suggestion
    CODE(0, LcnModuleCodeSubHandler.class, LcnModuleCodeSubHandler::new);

    private final int count;
    private final Class<? extends AbstractLcnModuleSubHandler> subHandlerClass;
    private final BiFunction<LcnModuleHandler, ModInfo, ? extends AbstractLcnModuleSubHandler> handlerFactory;

    private <T extends AbstractLcnModuleSubHandler> LcnChannelGroup(int count, Class<T> subHandlerClass, 
        BiFunction<LcnModuleHandler, ModInfo, T> handlerFactory) {
        this.count = count;
        this.subHandlerClass = subHandlerClass;
        this.handlerFactory = handlerFactory;
    }
```